js/data/create_bucket.js

```diff
@@ -19,6 +19,21 @@ function createBucket(layer, buffers, z, overscaling, collisionDebug) {
         layout[k] = values[k].calculate(z, fakeZoomHistory);
     }
 
+    if (layer.type === 'symbol') {
+        // To reduce the number of labels that jump around when zooming we need
+        // to use a text-size value that is the same for all zoom levels.
+        // This calculates text-size at a high zoom level so that all tiles can
+        // use the same value when calculating anchor positions.
+        if (values['text-size']) {
+            layout['text-max-size'] = values['text-size'].calculate(18, fakeZoomHistory);
+            layout['text-size'] = values['text-size'].calculate(z + 1, fakeZoomHistory);
+        }
+        if (values['icon-size']) {
+            layout['icon-max-size'] = values['icon-size'].calculate(18, fakeZoomHistory);
+            layout['icon-size'] = values['icon-size'].calculate(z + 1, fakeZoomHistory);
+        }
```

Comment:
Is this going to work in all cases? What if there's a big discontinuity in the `text-size` function?